### PR TITLE
Update typescript interface function definion and bump version to 2.0.0

### PR DESCRIPTION
**Pull request by v9n**
Follow Step 2 and 3

```
npm install
cd packages/api-augment
npm run clean:defs
npm run generate
```

then run:

```
npm run clean
npm run build
```

Note: I changed the port 8855 in the download-metadata script to my local 2.0.0 rpc node port 9946

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "oak.js",
-  "version": "1.9.0-rc.4",
+  "version": "2.0.0-rc.4",
```
Comment (chrisli30) [inline package.json:3]: This version needs to be reset to "2.0.0-rc.1", meaning release candidate 1 for 2.0.0

File: packages/api-augment/src/interfaces/augment-api-events.ts [hunk 2]
```diff
@@ -48,64 +48,65 @@ declare module '@polkadot/api-base/types/events' {
...
+      CallCannotBeDecoded: AugmentedEvent<ApiType, [who: AccountId32, taskId: Bytes], { who: AccountId32, taskId: Bytes }>;
       /**
        * The result of the DynamicDispatch action.
        **/
-      DynamicDispatchResult: AugmentedEvent<ApiType, [who: AccountId32, taskId: H256, result: Result<Null, SpRuntimeDispatchError>], { who: AccountId32, taskId: H256, result: Result<Null, SpRuntimeDispatchError> }>;
+      DynamicDispatchResult: AugmentedEvent<ApiType, [who: AccountId32, taskId: Bytes, result: Result<Null, SpRuntimeDispatchError>], { who: AccountId32, taskId: Bytes, result: Result<Null, SpRuntimeDispatchError> }>;
       /**
        * Notify event for the task.
        **/
       Notify: AugmentedEvent<ApiType, [message: Bytes], { message: Bytes }>;
```
Comment (chrisli30) [inline packages/api-augment/src/interfaces/augment-api-events.ts:63]: Isn’t this Notify in automationTime deleted?

File: packages/api-augment/src/interfaces/augment-api-events.ts [hunk 3]
```diff
@@ -48,64 +48,65 @@ declare module '@polkadot/api-base/types/events' {
...
-      SuccesfullyAutoCompoundedDelegatorStake: AugmentedEvent<ApiType, [taskId: H256, amount: u128], { taskId: H256, amount: u128 }>;
+      SuccesfullyAutoCompoundedDelegatorStake: AugmentedEvent<ApiType, [taskId: Bytes, amount: u128], { taskId: Bytes, amount: u128 }>;
```
Comment (chrisli30) [inline packages/api-augment/src/interfaces/augment-api-events.ts:64]: This should be deleted.

**Review by chrisli30 — CHANGES_REQUESTED** [review]
Hmm I don’t think the events in augment-api-events.ts is updated correctly. Either there’s no logic to pick up the deleted event names, or the local node is still returning them through rpc.

@v9n maybe try to completely delete this file before running the npm scripts?

**Comment by chrisli30** [comment]
@imstar15, the types package’s code is behind. How do we update the `types` package?
CleanShot 2023-08-04 at 19 57 39

**Comment by imstar15** [comment]
> @imstar15, the types package’s code is behind. How do we update the `types` package? CleanShot 2023-08-04 at 19 57 39

This line of configuration also needs to be manually updated to 2.0.0-rc.1.
packages/api-augment/package.json also needs to be manually updated to 2.0.0-rc.1.

File: package-lock.json [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
   "name": "oak.js",
-  "version": "1.9.0-rc.4",
+  "version": "2.0.0-rc.4",
   "lockfileVersion": 2,
   "requires": true,
   "packages": {
     "": {
       "name": "oak.js",
-      "version": "1.9.0-rc.4",
+      "version": "2.0.0-rc.4",
```
Comment (chrisli30) [inline package-lock.json:9]: Discrepancy, `2.0.0-rc.1` but not `2.0.0-rc.4`. The same applies to the line 3.

**Review by chrisli30 — APPROVED** [review]
The changes are published on npm as v2.0.0-rc.1 as tested in xcm-demo code. Good to merge.

@v9n A couple of things to keep in mind of.
1. The commit message `regen` is too simple. Please ALWAYS make sure the message explains the reason a code commit. For example, in the case you can use `Regenerate the api-augment files with the correct oak-blockchain version v2.0.0`
2. The previous version of this PR was incorrect. I assumed that the regenerate scripts ran against an old blockchain code version.

**Review by chrisli30 — APPROVED** [review]

**Pull request merged**